src/components/AnonymousChat/child/ChatItemAttachment.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-unexpected-multiline */
+import * as React from 'react';
+import {Linking, Text, TouchableOpacity, View} from 'react-native';
+import FastImage from 'react-native-fast-image';
+import {useNavigation} from '@react-navigation/core';
+
+import IconFile from '../../../assets/icon/IconFile';
+import IconVideoPlay from '../../../assets/icon/IconVideoPlay';
+import {formatBytes} from '../../../utils/string/StringUtils';
+import {ChatItemAttachmentStyles} from './ChatItemAttachmentStyles';
+
+const ChatItemAttachment = ({attachments = []}: any) => {
+  const navigation = useNavigation();
+
+  const onOpenMediaPreview = (medias, index) => {
+    if (medias.find((media) => media.type === 'video')) {
+      navigation.push('VideoViewer', {
+        title: 'Video',
+        url: medias[index].video_path
+      });
+    } else {
+      navigation.push('ImageViewer', {
+        title: 'Photo',
+        index,
+        images: medias
+          .filter((media) => media.type === 'image')
+          .map((media) => ({url: media.asset_url}))
+      });
+    }
+  };
+
+  return (
+    <View
+      style={[
+        ChatItemAttachmentStyles.attachmentContainer,
+        attachments?.find((item) => item.type === 'file') ? {height: 'auto'} : {}
+      ]}>
+      {attachments
+        .filter((item, index) => index <= 3)
+        .map((item, index) =>
+          item.type === 'file' ? (
+            <TouchableOpacity
+              key={item.file_path}
+              style={{flex: 1}}
+              activeOpacity={1}
+              onPress={() => Linking.openURL(item.file_path)}>
+              <View style={ChatItemAttachmentStyles.attachmentFileContainer}>
+                <View style={ChatItemAttachmentStyles.attachmentFileContent}>
+                  <Text style={ChatItemAttachmentStyles.attachmentFileName} numberOfLines={1}>
+                    {item.file_name}
+                  </Text>
+                  <View>
+                    <Text style={ChatItemAttachmentStyles.attachmentFileInfo}>
+                      {formatBytes(item.file_size)} •{' '}
+                      {item.file_name
+                        ?.split('.')
+                        [item.file_name?.split('.')?.length - 1]?.toUpperCase()}
+                    </Text>
+                  </View>
+                </View>
+                <View style={ChatItemAttachmentStyles.attachmentFileIcon}>
+                  <IconFile />
+                </View>
+              </View>
+            </TouchableOpacity>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code for rendering file attachments is quite complex and could be extracted into a separate component to improve readability and maintainability. 



<!-- This is an auto-generated comment by OSS CodeRabbit -->